server/swap/swap.go

```diff
@@ -840,20 +840,14 @@ func (s *Swapper) processInit(msg *msgjson.Message, params *msgjson.Init, stepIn
 		return coinwaiter.TryAgain
 	}
 
-	// Decode the contract and audit the contract.
-	recipient, val, err := coin.AuditContract()
-	if err != nil {
-		s.respondError(msg.ID, actor.user, msgjson.ContractError, fmt.Sprintf("error auditing contract: %v", err))
-		return coinwaiter.DontTryAgain
-	}
-	if recipient != counterParty.order.Trade().SwapAddress() {
+	if contract.Address() != counterParty.order.Trade().SwapAddress() {
 		s.respondError(msg.ID, actor.user, msgjson.ContractError,
-			fmt.Sprintf("incorrect recipient. expected %s. got %s", recipient, counterParty.order.Trade().SwapAddress()))
+			fmt.Sprintf("incorrect recipient. expected %s. got %s", contract.Address(), counterParty.order.Trade().SwapAddress()))
 		return coinwaiter.DontTryAgain
 	}
-	if val != stepInfo.checkVal {
+	if contract.Value() != stepInfo.checkVal {
 		s.respondError(msg.ID, actor.user, msgjson.ContractError,
-			fmt.Sprintf("contract error. expected contract value to be %d, got %d", stepInfo.checkVal, val))
+			fmt.Sprintf("contract error. expected contract value to be %d, got %d", stepInfo.checkVal, contract.Value()))
 		return coinwaiter.DontTryAgain
 	}
```

Comment:
Since part of the contract auditing occurs from `Contract` up above (https://github.com/decred/dcrdex/pull/202/files#diff-5f1d20749025f318cd1a8ba0915ebd37R833), the returned `err` from `Contract` could indicate structural problems with the swap contract, which would not warrant a TryAgain.